### PR TITLE
[ISSUE #3521] task6 add integration test for shared thread pool

### LINKED ISSUE #3521: [Task] add integration tests for shared thread pool
### Description

As a subtask of https://github.com/apache/incubator-shenyu/issues/3026, we need to add some integration tests to make sure that shared thread pool is working fine.

### Task List

1. - [x] add integration test in `shenyu-integrated-test-alibaba-dubbo`; @renzhuyan
2. - [x] add integration test in `shenyu-integrated-test-apache-dubbo`; @dragon-zhang 
3. - [x] add integration test in `shenyu-integrated-test-grpc`; @MichaelDeSteven
4. - [x] add integration test in `shenyu-integrated-test-motan`; @sunshujie1990 
5. - [x] add integration test in `shenyu-integrated-test-sofa`; @ShawnSiao
6. - [ ] add integration test in `shenyu-integrated-test-combination`; @dragon-zhang

### PR DESCRIPTION
**Pull request by dragon-zhang**
Fixes #3521 

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        Type returnType = new TypeToken<String>() {
+        }.getType();
+        String response = HttpHelper.INSTANCE.postGateway("/motan/demo/hello", request, returnType);
+        assertEquals("hello shenyu", response);
+    }
+    
+    @Test
+    public void testSofa() {
+        try {
+            SofaTestData response = HttpHelper.INSTANCE.getFromGateway("/sofa/findById?id=1001", new TypeToken<SofaTestData>() {
```
Comment (JooKS-me) [inline shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java:101]: There is no problem in other places, that is, the catch is used here, and I see that the sofa test in the log will make an error. I suggest to solve the problem of the plug-in first and then merge the pr.

File: shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        String dubbo = HttpHelper.INSTANCE.getFromGateway("/shenyu/getFromDubbo", String.class);
+        assertEquals(spring, dubbo);
+        String grpc = HttpHelper.INSTANCE.getFromGateway("/shenyu/getFromGrpc", String.class);
+        assertEquals(spring, grpc);
+        String motan = HttpHelper.INSTANCE.getFromGateway("/shenyu/getFromMotan", String.class);
+        assertEquals(spring, motan);
+        try {
+            String sofa = HttpHelper.INSTANCE.getFromGateway("/shenyu/getFromSofa", String.class);
+            assertEquals(spring, sofa);
+        } catch (Throwable e) {
```
Comment (JooKS-me) [inline shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java:122]: There is no problem in other places, that is, the catch is used here, and I see that the sofa test in the log will make an error. I suggest to solve the problem of the plug-in first and then merge the pr.

Comment (dragon-zhang) [reply shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java:122]: sure

**Review by yu199195 — APPROVED** [review]

File: shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java (hunk 1 above)
Comment (dragon-zhang) [reply shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java:101]: em...I tried at best, exclude dependencies, upgrade sofa version, all not work. How about remove `sofa test part` from `SharedThreadPoolTest`? I think `#testApacheDubbo`, `#testGrpc`, `#testMotan` and `#testIsOneThreadPool` has proved enugth.

Comment (JooKS-me) [reply shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java:101]: Maybe, we can add a TODO comment here.

**Pull request merged**